db/fixtures/ae_datastore/ManageIQ/Cloud/Orchestration/Provisioning/StateMachines/Methods.class/__methods__/postprovision.rb

```diff
@@ -1,22 +1,51 @@
 #
 # Description: This method examines the orchestration stack provisioned
 #
-def dump_stack_outputs(stack)
-  $evm.log("info", "Outputs from stack #{stack.name}")
-  stack.outputs.each do |output|
-    $evm.log("info", "Key #{output.key}, value #{output.value}")
-  end
-end
+module ManageIQ
+  module Automate
+    module Cloud
+      module Orchestration
+        module Provisioning
+          module StateMachines
+            class PostProvision
+              def initialize(handle = $evm)
+                @handle = handle
+              end
 
-$evm.log("info", "Starting Orchestration Post-Provisioning")
+              def main
+                @handle.log("info", "Starting Orchestration Post-Provisioning")
 
-task = $evm.root["service_template_provision_task"]
-service = task.destination
-stack = service.orchestration_stack
+                task = @handle.root["service_template_provision_task"]
+                service = task.destination
 
-service.post_provision_configure
+                debug = @handle.inputs['debug'] || false
+                if debug
+                  stack = service.orchestration_stack
+                  dump_stack_outputs(stack)
+                end
 
-# You can add logic to process the stack object in VMDB
-# For example, dump all outputs from the stack
-#
-# dump_stack_outputs(stack)
+                service.post_provision_configure
+
+                # You can add logic to process the stack object in VMDB
+                # For example, dump all outputs from the stack
+                #
+                # dump_stack_outputs(stack)
```

Comment:
The comment section can be deleted since `dump_stack_outputs` is not controlled by the debug flag